tests/installation/upgrade_select.pm

```diff
@@ -39,7 +39,10 @@ sub run {
         send_key $cmd{next};
     }
     if (match_has_tag("select-for-update")) {
-        send_key $cmd{next};
+        assert_screen('select-for-update-arch');
+        if (match_has_tag("select-for-update-arch")) {
```

Comment:
Chenzi, 
You have more than one needles which have tag "select-for-update-arch",right?
if no, I suggest removing this line :)

Reply:
Yes, there are eight needles within this tag.